src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/JsonOutputWriter.java

```diff
@@ -26,22 +26,26 @@
 
 public class JsonOutputWriter extends OutputWriter {
 
+    public JsonOutputWriter(final Collection<OutputField> fields, final OutputStream outputStream,
+                            final boolean jsonEnvelopeEnabled) {
+        super(outputStream, new Builder().addFields(fields).build(jsonEnvelopeEnabled));
+    }
+
     public JsonOutputWriter(final Collection<OutputField> fields, final OutputStream outputStream) {
-        super(outputStream, new Builder().addFields(fields).build());
+        super(outputStream, new Builder().addFields(fields).build(true));
     }
 
     static final class Builder {
         private final JsonOutputFieldComposer fieldsComposer = new JsonOutputFieldComposer();
 
         final JsonOutputWriter.Builder addFields(final Collection<OutputField> fields) {
             Objects.requireNonNull(fields, "fields cannot be null");
-
             fieldsComposer.addFields(fields);
             return this;
         }
 
-        final JsonOutputStreamWriter build() {
-            return new JsonOutputStreamWriter(fieldsComposer.fieldBuilders);
+        final JsonOutputStreamWriter build(final boolean jsonEnvelopeEnabled) {
```

Comment:
To keep the builders in shape, `jsonEnvelopeEnabled` should probably be another `with` call